### PR TITLE
fix: fix columns width on long rowset

### PR DESCRIPTION
**Pull request by astandrik**
Closes https://github.com/ydb-platform/ydb-embedded-ui/issues/1340
Closes https://github.com/ydb-platform/ydb-embedded-ui/issues/1386
Closes https://github.com/ydb-platform/ydb-embedded-ui/issues/1387

[Stand](http://ydb-vla-dev02-005.search.yandex.net:8765/monitoring/tenant?tenantPage=query&queryTab=newQuery&diagnosticsTab=topQueries&topQueriesDateFrom=1721228339103&topQueriesDateTo=1721314739103&selectedConsumer=&name=%2Fdev02&clckid=5cd66065&schema=%2Fdev02%2F.sys%2Fds_vslots)

### Test Status: <span style="color: orange;">⚠️ FLAKY</span>
📊 [Full Report](https://ydb-platform.github.io/ydb-embedded-ui/1384/)

| Total | Passed | Failed | Flaky | Skipped |
|:-----:|:------:|:------:|:-----:|:-------:|
| 124 | 123 | 0 | 1 | 0 |

### Bundle Size: ✅
Current: 79.13 MB | Main: 79.13 MB
Diff: +0.00 MB (0.00%)

✅ Bundle size unchanged.

<details>
<summary>ℹ️ CI Information</summary>

- Test recordings for failed tests are available in the full report.
- Bundle size is measured for the entire 'dist' directory.
- 📊 indicates links to detailed reports.
- 🔺 indicates increase, 🔽 decrease, and ✅ no change in bundle size.
</details>

### CODE REVIEW

**Comment by astandrik** [comment]
Had a discussion with @adameat to analyze first 100 rows in order to calculate predicted width

File: src/components/QueryResultTable/QueryResultTable.tsx [hunk 1]
```diff
@@ -82,7 +96,7 @@ export const QueryResultTable = (props: QueryResultTableProps) => {
 
     const data = React.useMemo(() => prepareQueryResponse(rawData), [rawData]);
     const columns = React.useMemo(() => {
-        return rawColumns ? prepareTypedColumns(rawColumns) : prepareGenericColumns(data);
+        return rawColumns ? prepareTypedColumns(rawColumns, data) : prepareGenericColumns(data);
```
Comment (artemmufazalov) [inline src/components/QueryResultTable/QueryResultTable.tsx:91]: Please, add the same logic to `prepareGenericColumns`

File: src/components/QueryResultTable/utils/getColumnWidth.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    data,
+    name,
+    columnType,
+}: {
+    data?: KeyValueRow[];
+    name: string;
+    columnType?: string;
+}) => {
+    let maxColumnContentLength = name.length;
+    const headerPadding = columnType === 'number' ? 40 : 20;
```
Comment (artemmufazalov) [inline src/components/QueryResultTable/utils/getColumnWidth.ts:15]: What is `headerPadding`? Why it's different for numbers and other types?

**Review by artemmufazalov — CHANGES_REQUESTED** [review]

File: src/components/QueryResultTable/utils/getColumnWidth.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+            const cellLength = row[name] ? String(row[name]).length : 0;
+            maxColumnContentLength = Math.max(maxColumnContentLength, cellLength);
+
+            if (maxColumnContentLength * 10 + HEADER_PADDING >= MAX_COLUMN_WIDTH) {
+                return MAX_COLUMN_WIDTH;
+            }
+        }
+    }
+
+    return Math.min(maxColumnContentLength * 10 + HEADER_PADDING, MAX_COLUMN_WIDTH);
```
Comment (Raubzeug) [inline src/components/QueryResultTable/utils/getColumnWidth.ts:20]: it seems we may skip this check, cause it's already done: https://github.com/ydb-platform/ydb-embedded-ui/pull/1384/files#diff-372ca73fb142649c09d5b178debd37c1c6507e47eb54bf23cbf447d7f4904e1bR14

**Review by Raubzeug — CHANGES_REQUESTED** [review]

**Review by Raubzeug — APPROVED** [review]

**Review by artemmufazalov — APPROVED** [review]

**Pull request merged**